marklogic-data-hub/src/test/java/com/marklogic/hub/dhs/DeployUserAmpsTest.java

```diff
@@ -0,0 +1,96 @@
+package com.marklogic.hub.dhs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.impl.Versions;
+import com.marklogic.mgmt.resource.security.AmpManager;
+import com.marklogic.rest.util.ResourcesFragment;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.nio.file.Path;
+
+
+public class DeployUserAmpsTest extends AbstractHubCoreTest {
+    ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    void checkIfTestsCanBeRun(){
+        Versions.MarkLogicVersion mlVersion = versions.getMLVersion();
+        Assertions.assertTrue(mlVersion.isNightly() || (mlVersion.getMajor() > 10) ||(mlVersion.getMajor() == 10 && mlVersion.getMinor() >= 440));
+    }
+
+    @Test
+    void createAmpWithInheritableRole() {
+        //POST request
+        writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME, "data-hub-developer");
+        ResourcesFragment amps = new AmpManager(runAsDataHubSecurityAdmin().getManageClient()).getAsXml();
+        Assertions.assertTrue(amps.resourceExists("testAmp"));
+
+        //PUT request(doesn't work)
+        writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME,"data-hub-developer", "data-hub-operator");
+
+        runAsDataHubSecurityAdmin().getManageClient().delete("/manage/v2/amps/testAmp?namespace=&document-uri=/data-hub/5/impl/jobs.sjs&modules-database=data-hub-MODULES");
```

Comment:
This is fine - a user can only do an undeploy locally, and that will likely be as an admin user.